lib/Data/Enum.hs

```diff
@@ -40,23 +40,36 @@ boundedEnumFromThen n1 n2
     i_n1 = fromEnum n1
     i_n2 = fromEnum n2
 
+numericEnumFrom :: (Num a) => a -> [a]
+numericEnumFrom n = n : numericEnumFrom (n + 1)
+
+numericEnumFromThen :: (Num a) => a -> a -> [a]
+numericEnumFromThen n m = from n
+  where
+    d = m - n
+    from i = i : from (i + d)
+
+numericEnumFromTo :: (Num a, Ord a) => a -> a -> [a]
+numericEnumFromTo l h = takeWhile (<= h) (numericEnumFrom l)
```

Comment:
The report and `base` use `h + 1/2` here, to account for rounding errors, but since there are currently no `Enum` instances for float types, that shouldn't matter, so I kept it like this.

Reply:
Thank you!  I'll merge this ASAP.